monitoring/sensu_subscription.py

```diff
@@ -0,0 +1,159 @@
+    default: /etc/sensu/conf.d/subscriptions.json
+  backup:
+    description:
+      - Create a backup file (if yes), including the timestamp information so you
+      - can get the original file back if you somehow clobbered it incorrectly.
+    choices: [ 'yes', 'no' ]
+    required: false
+    default: no
+requirements: [ ]
+author: Anders Ingemann
+'''
+
+RETURN = '''
+reasons:
+    description: the reasons why the moule changed or did not change something
+    returned: success
+    type: list
+    sample: ["channel subscription was absent and state is `present'"]
+'''
+
+EXAMPLES = '''
+# Subscribe to the nginx channel
+- name: subscribe to nginx checks
+  sensu_subscription: name=nginx
+
+# Unsubscribe from the common checks channel
+- name: unsubscribe from common checks
+  sensu_subscription: name=common state=absent
+'''
+
+
+def sensu_subscription(module, path, name, state='present', backup=False):
+    changed = False
+    reasons = []
+
+    try:
+        import json
+    except ImportError:
+        import simplejson as json
+
+    try:
+        config = json.load(open(path))
+    except IOError, e:
+        if e.errno is 2:  # File not found, non-fatal
+            if state == 'absent':
+                reasons.append('file did not exist and state is `absent\'')
+                return changed, reasons
+            config = {}
+        else:
+            module.fail_json(msg=str(e))
+    except ValueError:
+        msg = '{path} contains invalid JSON'.format(path=path)
+        module.fail_json(msg=msg)
+
+    if 'client' not in config:
+        if state == 'absent':
+            reasons.append('`client\' did not exist and state is `absent\'')
+            return changed, reasons
+        config['client'] = {}
+        changed = True
+        reasons.append('`client\' did not exist')
+
+    if 'subscriptions' not in config['client']:
+        if state == 'absent':
+            reasons.append('`client.subscriptions\' did not exist and state is `absent\'')
+            return changed
+        config['client']['subscriptions'] = []
+        changed = True
+        reasons.append('`client.subscriptions\' did not exist')
+
+    if name not in config['client']['subscriptions']:
+        if state == 'absent':
+            reasons.append('channel subscription was absent')
+            return changed
+        config['client']['subscriptions'].append(name)
+        changed = True
+        reasons.append('channel subscription was absent and state is `present\'')
+    else:
+        if state == 'absent':
+            config['client']['subscriptions'].remove(name)
+            changed = True
+            reasons.append('channel subscription was present and state is `absent\'')
+
+    if changed and not module.check_mode:
+        if backup:
+            module.backup_local(path)
+        try:
+            open(path, 'w').write(json.dumps(config, indent=2) + '\n')
+        except IOError, e:
+            module.fail_json(msg=str(e))
```

Comment:
we normally want to return a more 'friendly' error message to the user, but leaving that for a follow up (same issue in a couple of places):

`'We could not write to the %s file: %s' % (config, str(e))`